### PR TITLE
Fix try-runtime build and add missing migrations

### PR DESCRIPTION
**Pull request by fgamundi**
`try-runtime` now fails if there's a mismatch between a pallet's latest vs onchain versions. This flagged `PolkadotXcm` and `XcmpQueue` from live dancebox.
Missing migrations has been added

### CODE REVIEW

File: runtime/dancebox/src/migrations.rs [hunk 1]
```diff
@@ -209,19 +210,56 @@ where
...
+impl<T> Migration for XcmpQueueMigration<T>
+where
+    T: cumulus_pallet_xcmp_queue::Config,
+{
+    fn friendly_name(&self) -> &str {
+        "MM_XcmpQueueMigration"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        cumulus_pallet_xcmp_queue::migration::Migration::<T>::on_runtime_upgrade()
```
Comment (girazoki) [inline runtime/dancebox/src/migrations.rs:237]: do we know for certain this puts the v3 on the pallet?

Comment (girazoki) [reply runtime/dancebox/src/migrations.rs:237]: because we probably need to set v2 first before doing this on_runtime_upgrade

Comment (fgamundi) [reply runtime/dancebox/src/migrations.rs:237]: Yes, even if onchain is v1, it applies first the migration to v2 and immediately after the migration to v3 https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/pallets/xcmp-queue/src/migration.rs#L36C1-L46C4

**Comment by tmpolaczyk** [comment]
Can you merge master? I just merged a big PR that includes migrations, so there will be conflicts.

**Review by girazoki — APPROVED** [review]

**Comment by fgamundi** [comment]
@tmpolaczyk done

**Pull request merged**